packages/dd-trace/src/datastreams/writer.js

```diff
@@ -0,0 +1,66 @@
+const pkg = require('../../../../package.json')
+const log = require('../log')
+const request = require('../exporters/common/request')
+const { URL, format } = require('url')
+const msgpack = require('msgpack-lite')
+const zlib = require('zlib')
+const codec = msgpack.createCodec({ int64: true })
+
+function makeRequest (data, url, cb) {
+  const options = {
+    path: '/v0.1/pipeline_stats',
+    method: 'POST',
+    headers: {
+      'Datadog-Meta-Lang': 'javascript',
+      'Datadog-Meta-Tracer-Version': pkg.version,
+      'Content-Type': 'application/msgpack',
+      'Content-Encoding': 'gzip'
+    }
+  }
+
+  options.protocol = url.protocol
+  options.hostname = url.hostname
+  options.port = url.port
+
+  log.debug(() => `Request to the intake: ${JSON.stringify(options)}`)
+
+  request(data, options, (err, res) => {
+    cb(err, res)
+  })
+}
+
+class DataStreamsWriter {
+  constructor (config) {
+    const { hostname = '127.0.0.1', port = 8126, url } = config
+    this._url = url || new URL(format({
+      protocol: 'http:',
+      hostname: hostname || 'localhost',
+      port
+    }))
+  }
+
+  flush (payload) {
+    if (!request.writable) {
+      log.debug(() => `Maximum number of active requests reached. Payload discarded: ${JSON.stringify(payload)}`)
+      return
+    }
+    const encodedPayload = msgpack.encode(payload, { codec })
+
+    zlib.gzip(encodedPayload, { level: 1 }, (err, compressedData) => {
```

Comment:
Why is this gzipped? Historically, we've avoided gzipping to the agent because the CPU overhead is not worth the bandwidth reduction on a local network.

Reply:
@piochelepiotr should have insight to the protocol

Reply:
With the protobuf sketches, the encoded payload can have a lot of 0, so that's why we chose to add a gzip in the first implementation.
I don't think it should have a big perf impact because:
1. We're using level 1 encoding
2. The amount of data sent is much much lower than traces, because we aggregate before sending.

Reply:
If it's not a lot of data, then why even bother with gzip?